src/components/search/SearchHeader2.tsx

```diff
@@ -0,0 +1,150 @@
+import styled from '@emotion/styled';
+import { useRouter } from 'next/router';
+import { useEffect, useRef } from 'react';
+import type { FormEvent } from 'react';
+import { DeleteIcon, SearchIcon } from 'assets/icons';
+import { Z_INDEX } from 'constants/styles';
+import { useSearchKeywordStorage, useDebounce } from 'hooks/common';
+import { SVGVerticalAlignStyle, theme } from 'styles';
+
+interface SearchHeaderProps {
+  from: string;
+  to: (path: string) => string;
+  saveToStorage?: boolean;
+  initialValue?: string | null;
+}
+
+export const SearchHeader2 = ({
+  from,
+  to,
+  saveToStorage,
+  initialValue,
+}: SearchHeaderProps) => {
+  const router = useRouter();
+
+  const searchRef = useRef<HTMLInputElement | null>(null);
+
+  const { handleSaveSearchKeyword } = useSearchKeywordStorage();
+
+  const handleClearSearchKeyword = () => {
+    const searchElement = searchRef.current;
+    if (!searchElement) return;
+
+    searchElement.value = '';
+    searchElement.focus();
```

Comment:
clear 할 때 빈문자열 push 해주셔야 하이라이트 제거됩니다. 확인 부탁드려요
```suggestion
    searchElement.focus();
    void router.push('', undefined, { shallow: true });
```